### PR TITLE
[NAE-1363] Refactor panel featured values

### PR DESCRIPTION
**Pull request by renczesstefan**
# Description

Multiple implementations existed of the same column resolution in task-panel and case-panel. Now, there is a new component created that implements the logic only once.

Implements [NAE-1363]

## Dependencies

No new dependencies were introduced.

### Third party dependencies

No new dependencies were introduced.

### Blocking Pull requests

Pull request #43 is needed to be finished first to upgrade the new component to Angular 13 paradigm.

## How Has Been This Tested?

This was tested manually and via unit tests.

### Test Configuration

Netgrif Frontend PR Config
| Name                | Tested on |
|---------------------| --------- |
| OS                  |   macOS Monterey 12.2.1        |
| Runtime             |  Node 12.22.10         |
| Dependency Manager  |  NPM 6.14.16         |
| Framework version   |  Angular 10.2.3         |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @minop
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1363]: https://netgrif.atlassian.net/browse/NAE-1363?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/netgrif-components-core/src/lib/panel/abstract/panel-with-header-binding.ts
Comment (minop) [inline projects/netgrif-components-core/src/lib/panel/abstract/panel-with-header-binding.ts:44]: the commented line can be removed, the rest can be achieved with a `.map()` method call

File: projects/netgrif-components-core/src/lib/panel/panel-item/abstract-panel-item.component.ts
Comment (minop) [inline projects/netgrif-components-core/src/lib/panel/panel-item/abstract-panel-item.component.ts:15]: why are the empty methods necessary?

File: projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html
Comment (minop) [inline projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html:2]: perhaps a better way would be to not provide the index, but a boolean that would directly specify weather the case icon should be displayed or not. That would make the component more readable in my opinion

File: projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html
Comment (minop) [inline projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html:16]: the `width-100` class is missing in the scss file of this component

File: projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html
Comment (minop) [inline projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html:2]: the `panel-primary-icon` class is missing from the scss file of this component

File: projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.ts
Comment (minop) [inline projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.ts:15]: is the empty method necessary?

File: projects/netgrif-components/src/lib/panel/panel.module.ts
Comment (minop) [inline projects/netgrif-components/src/lib/panel/panel.module.ts:28]: it might be a good idea to export the component as well to make creation of custom panels simpler

**Review by minop — CHANGES_REQUESTED** [review]

File: projects/netgrif-components-core/src/lib/panel/abstract/panel-with-header-binding.ts
Comment (renczesstefan) [reply projects/netgrif-components-core/src/lib/panel/abstract/panel-with-header-binding.ts:44]: It is solved now with map stream.

File: projects/netgrif-components-core/src/lib/panel/panel-item/abstract-panel-item.component.ts
Comment (renczesstefan) [reply projects/netgrif-components-core/src/lib/panel/panel-item/abstract-panel-item.component.ts:15]: Method will be removed.

File: projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html
Comment (renczesstefan) [reply projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html:2]: It is now refactored according to you.

File: projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html
Comment (renczesstefan) [reply projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html:16]: It is now added to the scss file.

File: projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html
Comment (renczesstefan) [reply projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html:2]: It is present in panel.theme.scss file that is a wrapper theme file for whole panel component. It is not even present in the task-panel.component.scss nor in case-panel.component.ts, that served as old solution for this panel featured values. Is it ok?

File: projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.ts
Comment (renczesstefan) [reply projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.ts:15]: Methods have been removed.

File: projects/netgrif-components/src/lib/panel/panel.module.ts
Comment (renczesstefan) [reply projects/netgrif-components/src/lib/panel/panel.module.ts:28]: It is exported in public.api.ts. Is ok for you?

File: projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html
Comment (minop) [reply projects/netgrif-components/src/lib/panel/panel-item/panel-item.component.html:2]: is the panel theme then imported into this component somehow? Does the class do anything in the browser?

File: projects/netgrif-components/src/lib/panel/panel.module.ts
Comment (minop) [reply projects/netgrif-components/src/lib/panel/panel.module.ts:28]: both is neccessary. If you export it in `public.api` then you can import the class, but if you want to use the component in your html templates it has to be exported in the module as well

**Review by minop — DISMISSED** [review]

**Review by minop — DISMISSED** [review]

**Comment by mladoniczky** [comment]
PR #43 is not blocking this PR, but changes done in this issue needs to be revisited after merging #43 .

**Review by mladoniczky — APPROVED** [review]

**Review by mladoniczky — APPROVED** [review]

**Pull request merged**
